File: backend/src/main/java/com/bloodline/bloodline_backend/entity/Sample.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Sample = sequelize.define('Sample', {
    sampleID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    surchargeID: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    bookingID: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    participantID: {
        type: DataTypes.INTEGER,
        allowNull: false,
        unique: true
    },
    sampleType: {
        type: DataTypes.STRING(100),
        allowNull: false
    },
    receivedDate: {
        type: DataTypes.DATE,
        allowNull: true
    },
    status: {
        type: DataTypes.STRING(100),
        allowNull: false
    }
}, {
    tableName: 'Sample',
    timestamps: false
});

module.exports = Sample; 

File: backend/src/main/java/com/bloodline/bloodline_backend/entity/Sample.java
package com.bloodline.bloodline_backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "Sample")
public class Sample {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "sampleID")
    private Integer sampleID;

    @ManyToOne
    @JoinColumn(name = "bookingID", nullable = false)
    private Booking booking;

    @ManyToOne
    @JoinColumn(name = "userID")
    private User user;

    @ManyToOne
    @JoinColumn(name = "participantID", nullable = false)
    private Participant participant;

    @Column(name = "typeOfCollection", nullable = false)
    private String typeOfCollection;

    @Column(name = "sampleType", nullable = false)
    private String sampleType;

    @Column(name = "receivedDate")
    private LocalDate receivedDate;

    @Column(name = "status", nullable = false)
    private String status;
} 